swap/swap.go

```diff
@@ -143,45 +145,80 @@ func New(logpath string, dbPath string, prvkey *ecdsa.PrivateKey, backendURL str
 	// initialize the balances store
 	stateStore, err := state.NewDBStore(filepath.Join(dbPath, "swap.db"))
 	if err != nil {
-		return nil, fmt.Errorf("swap init error: %s", err)
+		return nil, fmt.Errorf("swap init error: error initializing statestore: %v", err)
 	}
 	if disconnectThreshold <= paymentThreshold {
 		return nil, fmt.Errorf("swap init error: disconnect threshold lower or at payment threshold. DisconnectThreshold: %d, PaymentThreshold: %d", disconnectThreshold, paymentThreshold)
 	}
 	backend, err := ethclient.Dial(backendURL)
 	if err != nil {
-		return nil, fmt.Errorf("swap init error: error connecting to Ethereum API %s: %s", backendURL, err)
+		return nil, fmt.Errorf("swap init error: error connecting to Ethereum API %s: %v", backendURL, err)
 	}
+	networkID, err := backend.NetworkID(context.TODO())
+	if err != nil {
+		return nil, fmt.Errorf("swap init error: error retrieving networkID from backendURL: %v", err)
+	}
+	var usedBeforeAtNetwork uint64
+	var usedBefore bool
+	err = stateStore.Get(usedBeforeAtNetworkKey(networkID.String()), &usedBeforeAtNetwork)
```

Comment:
Why is this key per `networkId`. This should be a global value.

Reply:
Fixed